store/sqlstore/oauth_store.go

```diff
@@ -59,277 +59,234 @@ func (as SqlOAuthStore) CreateIndexesIfNotExists() {
 	as.CreateIndexIfNotExists("idx_oauthauthdata_client_id", "OAuthAuthData", "Code")
 }
 
-func (as SqlOAuthStore) SaveApp(app *model.OAuthApp) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if len(app.Id) > 0 {
-			result.Err = model.NewAppError("SqlOAuthStore.SaveApp", "store.sql_oauth.save_app.existing.app_error", nil, "app_id="+app.Id, http.StatusBadRequest)
-			return
-		}
+func (as SqlOAuthStore) SaveApp(app *model.OAuthApp) (*model.OAuthApp, *model.AppError) {
+	if len(app.Id) > 0 {
+		return nil, model.NewAppError("SqlOAuthStore.SaveApp", "store.sql_oauth.save_app.existing.app_error", nil, "app_id="+app.Id, http.StatusBadRequest)
+	}
 
-		app.PreSave()
-		if result.Err = app.IsValid(); result.Err != nil {
-			return
-		}
+	app.PreSave()
+	if err := app.IsValid(); err != nil {
+		return nil, err
+	}
 
-		if err := as.GetMaster().Insert(app); err != nil {
-			result.Err = model.NewAppError("SqlOAuthStore.SaveApp", "store.sql_oauth.save_app.save.app_error", nil, "app_id="+app.Id+", "+err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = app
-		}
-	})
+	if err := as.GetMaster().Insert(app); err != nil {
+		return nil, model.NewAppError("SqlOAuthStore.SaveApp", "store.sql_oauth.save_app.save.app_error", nil, "app_id="+app.Id+", "+err.Error(), http.StatusInternalServerError)
+	}
+	return app, nil
 }
 
-func (as SqlOAuthStore) UpdateApp(app *model.OAuthApp) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		app.PreUpdate()
+func (as SqlOAuthStore) UpdateApp(app *model.OAuthApp) (*model.OAuthApp, *model.AppError) {
```

Comment:
I changed the behavior here on purpose, I returning just the new value, not the old one because wasn't used anywhere, and is easier to understand.